LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -302,7 +302,67 @@ public void AssignInstructorsToClassrooms()
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            int count = 0;
+            string action = "Register students for classes";
+            ReportHeader(actionName: action);
+
+            if (ClassroomDictionary.Count == 0)
+            {
+                ReportError(missingResource: "Classroom", failedAction: action);
+            }
+            else if (ActiveStudentList.Count == 0)
+            {
+                ReportError(missingResource: "Student", failedAction: action);
+            }
+            else
+            {
+                foreach (Classroom classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        count++;
+
+                        ReportItem(itemName: classroom.ToString(), actionName: action, itemIndex: count);
+                    }
+                }
+            }
+
+            ReportSummary(actionName: action, count: count);
+            ReportFooter(actionName: action);
+        }
+
+        void UpdateStudentList(Classroom classroom)
```

Comment:
This is a helper function, you don't need header, summary and footer here, just use ReportItem when you register a student.